client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -305,6 +305,22 @@ Undocumented.prototype.getJetpackConnectionStatus = function( siteId, fn ) {
 	}, fn );
 };
 
+/*
+ * Retrieve current user's connection data for a Jetpack site.
+ *
+ * @param {int}      [siteId]
+ * @param {Function} fn
+ * @api public
+ */
+Undocumented.prototype.getJetpackUserConnectionData = function( siteId, fn ) {
```

Comment:
Did you consider using the data-layer? One of the immediate benefits is to make `isRequesting` logic redundant.